src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -73,6 +75,13 @@ AttributeSyntax attribute
 		);
 
 	private SyntaxToken RemoveAsyncSuffix( SyntaxToken ident, bool optional = false ) {
+		if( m_removeAsyncAnywhere && ident.ValueText.Contains( "Async" ) ) {

```

Comment:
Should this check `optional`?

Reply:
I don't think so, because there are cases where a function may return IAsyncEnumerable and not end in Async and we want the sync generator to transform it without a diagnostic